src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -419,6 +419,19 @@ public Page GetPage(long pageNumber)
             return GetPageInternal(pageNumber);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Page GetPage(long pageNumber, PagerRef pagerRef)
+        {
+            if (_disposed)
+                ThrowObjectDisposed();
+
+            // PERF: ReadOnly pages will not have a PagerRef, so we can still use the cache. 
+            if (_pageLocator.TryGetReadOnlyPage(pageNumber, out Page result))
```

Comment:
Here you aren't assigning values to `pagerRef` which we need when running in 32 bits (in order to call `EnsureMapped` see my comments in 2e6393899fd)